llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -447,32 +473,44 @@ bool AIEHazardRecognizer::checkConflict(
 }
 
 void AIEHazardRecognizer::emitInScoreboard(const MCInstrDesc &Desc,
+                                           MemoryBankBits MemoryBanks,
                                            int DeltaCycles) {
   enterResources(
       Scoreboard, ItinData, Desc.getSchedClass(),
       getSlotSet(Desc, *TII->getFormatInterface(), IgnoreUnknownSlotSets),
-      DeltaCycles, FUDepthLimit);
+      MemoryBanks, TII->getFirstMemoryCycle(Desc.getSchedClass()), DeltaCycles,
+      FUDepthLimit);
 }
 
 void AIEHazardRecognizer::emitInScoreboard(
     ResourceScoreboard<FuncUnitWrapper> &TheScoreboard, const MCInstrDesc &Desc,
-    int DeltaCycles) const {
+    MemoryBankBits MemoryBanks, int DeltaCycles) const {
   enterResources(
       TheScoreboard, ItinData, Desc.getSchedClass(),
       getSlotSet(Desc, *TII->getFormatInterface(), IgnoreUnknownSlotSets),
-      DeltaCycles, FUDepthLimit);
+      MemoryBanks, TII->getFirstMemoryCycle(Desc.getSchedClass()), DeltaCycles,
+      FUDepthLimit);
 }
 
 void AIEHazardRecognizer::enterResources(
     ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
     const InstrItineraryData *ItinData, unsigned SchedClass, SlotBits SlotSet,
+    MemoryBankBits MemoryBanks, std::optional<int> MemoryAccessCycle,
     int DeltaCycles, std::optional<int> FUDepthLimit) {
   assert(Scoreboard.isValidDelta(DeltaCycles));
 
   // Append slot usage
   FuncUnitWrapper EmissionCycle(/*Req=*/0, /*Res=*/0, SlotSet);
   Scoreboard[DeltaCycles] |= EmissionCycle;
 
+  // Append memory bank usage
```

Comment:
I would be happy with an assert like `assert(MemoryBanks.emptry() || MemoryAccessCycle.has_value())`